zarr/v3/metadata.py

```diff
@@ -154,20 +163,19 @@ class ShardingCodecIndexLocation(Enum):
 
 
 @frozen
-class CoreArrayMetadata:
-    shape: ChunkCoords
+class ChunkMetadata:
```

Comment:
I'm thinking about the name for this class:

- the 'chunk" part.  None of the attributes of this class are specific to chunks per se -- if we aliased `chunk-shape` with `shape`, `data_type` to `dtype`, then we have a generic specification of a NDArray-like that wouldn't look out of place in numpy. So what if we drop the "chunk" references in the name and attributes?
The old name was `CoreArrayMetadata`. "core array metadata" feels like a better _description_ than "chunk metadata", but it's a bit long, and I don't think we need to semantically distinguish between "core" arrays and "non-core arrays (i.e., chunks)" in code like this. 

- the "metadata" part: I'm not sure that I would call the shape and dtype of an array "metadata" in the conventional sense -- "property" seems like a better fit? But i'm not sure. And we are already in a module called `metadata`, so repeating that word in class names might be redundant; what do you think of moving from the `<Thing>Metadata` pattern to `<Thing>`, as long as the "metadata-ness" is conveyed by the module name?

A couple of alternative names for this class, off the top of my head:
- `ArraySpec`
- `ArrayInfo`
- `ArrayProps`
- `NDArray` (I kind of prefer this one, as long as we namespace it in the right module to make things unambiguous)

Reply:
I would find `NDArray` confusing, because it is not an array but only contains information about an array. `ArraySpec` or `ArrayInfo` are fine with me.

Reply:
let's just go with `ArraySpec` and if people hate it we can find a better name at that time?